extensions/acc-core.md

```diff
@@ -0,0 +1,270 @@
+account registration.  If a valid callback is specified by the client during the `ACC REGISTER`
+command, the server MUST send the token to the specified callback.
+
+A `ACC VERIFY` command consists of the following format:
+
+    ACC VERIFY <accountname> <auth_code>
+
+The IRC server MAY forward the `ACC VERIFY` command to a central authority, or process it
+locally.
+
+Upon success, the IRC server MUST send the `RPL_VERIFYSUCCESS` numeric, which looks like:
+
+| No. | Label               | Format                                                                         |
+| --- | ------------------- | ------------------------------------------------------------------------------ |
+| 923 | `RPL_VERIFYSUCCESS` | `:<server> 923 <user_nickname> <accountname> :Account verification successful` |
+
+The IRC server MUST also send an `RPL_LOGGEDIN` (900) numeric and consider the client to be logged in to the
+account that has been successfully verified.
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these
+numerics:
+
+| No. | Label                             | Format                                                                   |
+| --- | --------------------------------- | ------------------------------------------------------------------------ |
+| 924 | `ERR_ACCOUNT_ALREADY_VERIFIED`    | `:<server> 924 <user_nickname> <accountname> :Account already verified`  |
+| 925 | `ERR_ACCOUNT_INVALID_VERIFY_CODE` | `:<server> 925 <user_nickname> <accountname> :Invalid verification code` |
+
+## The `ACCCOMMANDS` `RPL_ISUPPORT` token
+
+Implementations which provide these commands SHOULD signal their availability using the `ACCCOMMANDS`
+RPL_ISUPPORT (005) token.  The `ACCCOMMANDS` token takes a comma-separated list of supported
+commands.  Other extensions MAY include additional commands in the `ACCCOMMANDS` token than those
+documented here.
+
+A sample 005 reply indicating support for these commands is:
+
+    :irc.example.com 005 kaniini ACCCOMMANDS=REGISTER,VERIFY :are supported by this server
+
+## The `REGCALLBACKS` `RPL_ISUPPORT` token
+
+Implementations which require a verification token should list the types of verification callbacks
+they support, using the `REGCALLBACKS` RPL_ISUPPORT (005) token.  The `REGCALLBACKS` token takes
+a comma-separated list of supported commands.  In general, the `REGCALLBACKS` token is limited to
+URI schemes defined in the IANA URI Schemes registry, per [RFC 7595][rfc7595].
+
+  [rfc7595]: https://tools.ietf.org/html/rfc7595
+
+In the event that verification is not required by the implementation, the `REGCALLBACKS` token should
+be an empty value.
+
+A sample 005 reply indicating support for verification callbacks is:
+
+    :irc.example.com 005 kaniini REGCALLBACKS=mailto,sms :are supported by this server
+
+A sample 005 reply indicating that no verification callback methods are supported is:
+
+    :irc.example.com 005 kaniini REGCALLBACKS= :are supported by this server
```

Comment:
Surely there should be a way to specify that the server offers registration without callback but also offers e.g. mailto? That way a server can allow accounts to be e.g. tied to an email address or not, depending on what the user chooses?

e.g. `:irc.example.com 005 kaniini REGCALLBACKS=*,mailto :are supported by this server`

Reply:
Eh, sure. I'd be fine with this. Haven't dine any work on ACC for ages because of the lack of implementation interest, but when I do update I'll include this.